src/particles/diagnostics/ReducedBeamCharacteristics.cpp

```diff
@@ -101,25 +77,21 @@ namespace impactx::diagnostics
                 return {p_w,
                         p_x_mean, p_y_mean, p_t_mean,
                         p_px_mean, p_py_mean, p_pt_mean,
-                        p_x, p_x,
-                        p_y, p_y,
-                        p_t, p_t,
-                        p_px, p_px,
-                        p_py, p_py,
-                        p_pt, p_pt};
+                        p_x, p_y, p_t, p_px, p_py, p_pt,
+                        p_x, p_y, p_t, p_px, p_py, p_pt};
             },
-            reduce_ops
+            reduce_ops_1
         );
 
-        std::vector<amrex::ParticleReal> values_per_rank_1st = {
-                amrex::get<0>(r), // w
-                amrex::get<1>(r), // x_mean
-                amrex::get<2>(r), // y_mean
-                amrex::get<3>(r), // t_mean
-                amrex::get<4>(r), // px_mean
-                amrex::get<5>(r), // py_mean
-                amrex::get<6>(r), // pt_mean
-        };
+        std::vector<amrex::ParticleReal> values_per_rank_1st(num_red_ops_1_sum);
+
+        /* contains in this order:
+         * w, x_mean, y_mean, t_mean
+         * px_mean, py_mean, pt_mean
+         */
+        amrex::constexpr_for<0, num_red_ops_1_sum> ([&](auto i) {
+            values_per_rank_1st[i] = amrex::get<i>(r1);
```

Comment:
I see this in HiPACE++:
```suggestion
            values_per_rank_1st[i.value] = amrex::get<i.value>(r1);
```